### PR TITLE
Fix #848 - new [SQLFeatureStore] Deleting features fails for certain feature type prefix configurations

### LINKED ISSUE #848: [SQLFeatureStore] Deleting features fails for certain feature type prefix configurations
We observed that a delete by feature id against an SQLFeatureStore fails in some cases. This issue seems to be related to certain feature prefix configuration and differs between Java versions, e.g. we have two feature types with the following feature id prefices:

* APP_FEATURE_
* APP_FEATURE_X

We found that deleting the feature with id "APP_FEATURE_X_1" works in a Java 7 setup (tested with OpenJDK 7, 1.7.0_151 on Ubuntu 14.04 AMD64), while it fails with an exception on Java 8 (same machine, Oracle JDK 8, 1.8.0_144).

A first analysis points to method analyze(String) in class org.deegree.feature.persistence.sql.id.IdAnalyzer:

<pre>
...
        // TODO implement this more efficiently
        for ( String prefix : prefixToFt.keySet() ) {
            if ( featureOrGeomId.startsWith( prefix ) ) {
                ft = prefixToFt.get( prefix );
                break;
            }
        }
...
</pre>

We believe this is caused by the different key order in the respective HashMap implementations:

https://stackoverflow.com/questions/38587711/hashmap-keyset-returns-keys-in-different-order-in-java-7-java-8

### PR DESCRIPTION
**Pull request by nitram509**
Hi,

this PR supersedes #850.
It reduces the PR to the actual changed files and relevant commits.

Fore actual content details, please have a look at https://github.com/deegree/deegree3/pull/850

Regards
Martin

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -12,4 +12,4 @@ script: "travis_retry mvn verify -Pintegration-tests -B -V"
 env: MAVEN_OPTS="-XX:MaxPermSize=256m"
 cache:
   directories:
-    - $HOME/.m2/repository/
\ No newline at end of file
+    - $HOME/.m2/repository/
```
Comment (tfr42) [inline .travis.yml]: Can you please remove this change!

**Comment by copierrj** [comment]
Thank you for your contribution. The TMC meeting noticed that your PR contains a change to the .travis.yml file. It was therefore decided that we would to ask you if you can remove this change.

**Comment by nitram509** [comment]
@tfr42 sure, I'm willing to support and follow your advice.
BUT: I hesitate to do it, because I believe there is a miss-understanding.

This change was introduced 9/Feb/2015 and I did not touched/changed this line. I did merge from MASTER and there you can see, this change is in.
I believe there is a visualization issue in the Github Web-UI, which wrongly shows that I have added this line - which is not the case.

Indeed, I removed a line from .travis.ml
>>> dist: precise
(according to comment in #850).

Would you please double check and confirm,
that I should remove the caching configuration for the Maven repo-folder?

**Comment by tfr42** [comment]
> Would you please double check and confirm,

Looks ok. It seems to be a CR/LF character issue.

> that I should remove the caching configuration for the Maven repo-folder?

No, please leave it as it is.

**Pull request merged**